src/connectionconfig/connectionconfig.ts

```diff
@@ -31,6 +38,51 @@ export class ConnectionConfig implements IConnectionConfig {
         this._vscodeWrapper = value;
     }
 
+    private async assignMissingIds(): Promise<void> {
+        // Connection groups
+        const groups: IConnectionGroup[] = this.getGroupsFromSettings();
+
+        // ensure each group has an id
+        groups.forEach((group) => {
+            if (!group.id) {
+                group.id = Utils.generateGuid();
+            }
+        });
+
+        // ensure ROOT group exists
+        let rootGroup = groups.find(
+            (group) => group.name === this.RootGroupName,
+        );
+        if (!rootGroup) {
+            rootGroup = {
+                name: this.RootGroupName,
+                id: Utils.generateGuid(),
+            };
+            groups.push(rootGroup);
+        }
+
+        // Connection profiles
+        const profiles: IConnectionProfile[] = this.getProfilesFromSettings();
+
+        // ensure each profile has an id
+        profiles.forEach((profile) => {
+            ConnectionProfile.addIdIfMissing(profile);
+        });
+
+        // ensure each profile is in a group
+        profiles.forEach((profile) => {
```

Comment:
You could probably just use the previous forEach to also ensure that each profile is in a group.